src/libasr/asr_verify.cpp

```diff
@@ -595,7 +595,13 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
                 || is_a<Function_t>(*x.m_v) || is_a<ASR::EnumType_t>(*x.m_v),
             "Var_t::m_v " + x_mv_name + " does not point to a Variable_t, ExternalSymbol_t, " \
             "Function_t, Subroutine_t or EnumType_t");
-        require(symtab_in_scope(current_symtab, x.m_v),
+        SymbolTable *enum_symtab = nullptr;
+        if ( current_symtab->resolve_symbol("enum") != nullptr ) {
+            ASR::symbol_t *enum_s = current_symtab->resolve_symbol("enum");
+            enum_symtab = ASR::down_cast<ASR::EnumType_t>(enum_s)->m_symtab;
```

Comment:
Or I think we can add a check here:
```suggestion
            if (ASR::is_a<ASR::EnumType_t>(*enum_s)) {
                enum_symtab = ASR::down_cast<ASR::EnumType_t>(enum_s)->m_symtab;
            }
```